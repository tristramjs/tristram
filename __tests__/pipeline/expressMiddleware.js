import httpMocks from 'node-mocks-http';
import tristramMiddleware from '../../src/pipeline/expressMiddleware';

import Main from '../../src/main';
import PlainDataFetcher from '../../src/fetching/plainData';
import PlainFormatter from '../../src/formatting';

describe('expressMiddleware', () => {
	const data = [ { loc: 'http://url1.bar' }, { loc: 'http://url2.bar' }, { loc: 'http://url3.bar' } ];
	const fetcher = new PlainDataFetcher({ data });
	const formatter = new PlainFormatter({
		options: {
			hostname: 'localhost',
			cacheTime: 500,
			maxItemsPerSitemap: 2,
			path: 'sitemaps',
		},
	});
	const main = new Main({ fetchers: [ fetcher ], formatter, options: { hostname: '', cacheTime: 0 } });
	const middleware = tristramMiddleware(main);
	test('call next() if method is not GET', () => {
		const req = httpMocks.createRequest({ method: 'POST' });
		const res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next).then(() => expect(next).toHaveBeenCalled());
	});
	test('call next() if requested sitemap is not availabel', () => {
		const req = httpMocks.createRequest({ method: 'GET', url: 'path/to/sitemap-3.xml' });
		const res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next).then(() => expect(next).toHaveBeenCalled());
	});
	test('call next() if request is not for sitemap or index sitemap', () => {
		const req = httpMocks.createRequest({ method: 'GET', url: 'path/to/something-1.xml' });
		const res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next).then(() => expect(next).toHaveBeenCalled());
	});
	test('return indexSitemap if valid indexSitemap request', async () => {
		const req = httpMocks.createRequest({ method: 'GET', url: 'path/to/index.xml' });
		let res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next);
		expect(res._getData()).toMatchSnapshot();
	});
	test('return sitemap if valid sitemap request', () => {
		const req = httpMocks.createRequest({ method: 'GET', url: 'path/to/sitemap-1.xml' });
		let res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next);
		expect(res._getData()).toMatchSnapshot();
	});
	test('call next() if indexsitemap is requested but not availabel', () => {
		const formatter = new PlainFormatter({
			options: {
				hostname: 'localhost',
				cacheTime: 500,
				maxItemsPerSitemap: 10,
				path: 'sitemaps',
			},
		});
		const main = new Main({ fetchers: [ fetcher ], formatter, options: { hostname: '', cacheTime: 0 } });
		const middleware = tristramMiddleware(main);
		const req = httpMocks.createRequest({ method: 'GET', url: 'path/to/index.xml' });
		let res = httpMocks.createResponse();
		const next = jest.fn();
		middleware(req, res, next).then(() => expect(next).toHaveBeenCalled());
	});
});
