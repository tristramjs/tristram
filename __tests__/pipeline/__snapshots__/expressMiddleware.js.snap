// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expressMiddleware return indexSitemap if valid indexSitemap request 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><sitemapindex xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\"><sitemap><loc>https://localhost/sitemaps/sitemap-2.xml</loc></sitemap><sitemap><loc>https://localhost/sitemaps/sitemap-1.xml</loc></sitemap></sitemapindex>"`;

exports[`expressMiddleware return sitemap if valid sitemap request 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>http://url1.bar</loc></url><url><loc>http://url2.bar</loc></url></urlset>"`;
